v1/python-sdk/tutorials/automl-with-azureml/forecasting-demand-forecasting-many-models/auto-ml-forecasting-demand-forecasting-many-models.ipynb

```diff
@@ -0,0 +1,1626 @@
+    "\n",
+    "metrics_per_grain_day = compute_all_metrics(\n",
+    "    fcst_df=backtest,\n",
+    "    actual_col=target_column_name,\n",
+    "    fcst_col=\"predicted\",\n",
+    "    ts_id_colnames=[GRAIN_COL, PERIOD_COLUMN],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\n",
+    "    f'List of available metrics: {metrics_per_grain_day[\"metric_name\"].unique()}\\n---'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DESIRED_METRIC_NAME = \"root_mean_squared_error\"\n",
+    "\n",
+    "metrics_per_grain_day = metrics_per_grain_day.query(\n",
+    "    f'metric_name == \"{DESIRED_METRIC_NAME}\"'\n",
+    ")\n",
+    "metrics_per_grain_day[[GRAIN_COL, PERIOD_COLUMN]] = metrics_per_grain_day[\n",
+    "    \"time_series_id\"\n",
+    "].str.split(\"|\", 1, expand=True)\n",
+    "metrics_per_grain_day.to_csv(\n",
+    "    os.path.join(OUTPUT_DIR, \"metrics-automl.csv\"), index=False\n",
+    ")\n",
+    "metrics_per_grain_day.groupby(GRAIN_COL)[\"metric\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# baseline metrics\n",
+    "metrics_per_grain_day_base = compute_all_metrics(\n",
+    "    fcst_df=backtest,\n",
+    "    actual_col=target_column_name,\n",
+    "    fcst_col=\"predicted_base\",\n",
+    "    ts_id_colnames=[GRAIN_COL, PERIOD_COLUMN],\n",
+    ")\n",
+    "metrics_per_grain_day_base = metrics_per_grain_day_base.query(\n",
+    "    f'metric_name == \"{DESIRED_METRIC_NAME}\"'\n",
+    ")\n",
+    "metrics_per_grain_day_base[[GRAIN_COL, PERIOD_COLUMN]] = metrics_per_grain_day[\n",
+    "    \"time_series_id\"\n",
+    "].str.split(\"|\", 1, expand=True)\n",
+    "metrics_per_grain_day_base.to_csv(\n",
+    "    os.path.join(OUTPUT_DIR, \"metrics-base.csv\"), index=False\n",
+    ")\n",
+    "metrics_per_grain_day_base.groupby(GRAIN_COL)[\"metric\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 7.4  Visualize metrics\n",
+    "\n",
+    "In this section we plot metric evolution over time for the best AutoML and the baseline models."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "metrics_df = metrics_per_grain_day.drop(\"time_series_id\", axis=1).merge(\n",
+    "    metrics_per_grain_day_base.drop(\"time_series_id\", axis=1),\n",
+    "    on=[\"metric_name\", GRAIN_COL, PERIOD_COLUMN],\n",
+    "    how=\"inner\",\n",
+    "    suffixes=[\"\", \"_base\"],\n",
+    ")\n",
+    "metrics_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grain = [GRAIN_COL]\n",
+    "plot_filename = \"metrics_plot.pdf\"\n",
```

Comment:
Let us add the markdown cell above stating: Save the metrics evolution plots to pdf file.

Reply:
done.